src/services/userService.js

```diff
@@ -0,0 +1,45 @@
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: Math.floor(Math.random() * 1000),
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  users = users.filter((person) => person.id !== +id);
+};
```

Comment:
Code Quality: Your deleteUser function does not return any value. It would be helpful to return a value indicating whether the deletion was successful or not. For example, you could return the deleted user if the deletion was successful, or 'null' if no user was found with the given id.